base/workerpool.jl

```diff
@@ -0,0 +1,74 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+
+type WorkerPool
+    channel::RemoteChannel{Channel{Int}}
+end
+
+
+"""
+    WorkerPool([workers])
+
+Create a WorkerPool from a vector of worker ids.
+"""
+function WorkerPool(workers=Int[])
+
+    # Create a shared queue of available workers...
+    pool = WorkerPool(RemoteChannel(()->Channel{Int}(128)))
+
+    # Add workers to the pool...
+    for w in workers
+        put!(pool, w)
+    end
+
+    return pool
+end
+
+
+put!(pool::WorkerPool, w::Int) = put!(pool.channel, w)
+put!(pool::WorkerPool, w::Worker) = put!(pool.channel, w.id)
+
+isready(pool::WorkerPool) = isready(pool.channel)
+
+take!(pool::WorkerPool) = take!(pool.channel)
+
+
+"""
+    remotecall_fetch(f, pool::WorkerPool, args...)
+
+Call `f(args...)` on one of the workers in `pool`.
+"""
+function remotecall_fetch(f, pool::WorkerPool, args...)
+    worker = take!(pool)
+    try
+        remotecall_fetch(f, worker, args...)
+    finally
+        put!(pool, worker)
+    end
+end
+
+
+"""
+    default_worker_pool
+
+Built-in WorkerPool (used by `remote(f)`).
+"""
+_default_worker_pool = Nullable{WorkerPool}()
+function default_worker_pool()
+    if isnull(_default_worker_pool) && myid() == 1
+        set_default_worker_pool(WorkerPool())
+    end
+    return get(_default_worker_pool)
+end
+
+function set_default_worker_pool(p::WorkerPool)
+    global _default_worker_pool = Nullable(p)
+end
+
+
+"""
+    remote(f)
```

Comment:
Unless there is a solid convention that functions that return a function have an `f` suffix, I'd rather not change it as this point and introduce new conventions on the fly. It has been `remote` as proposed in https://github.com/JuliaLang/julia/issues/14843 since Jan without objection.

I'd be happy for it to be changed in a later functions-that-return-functions naming cleanup PR.
(There is probably an argument for a functions-that-return-iterators naming convention too...)